packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue

```diff
@@ -120,10 +119,6 @@ export default defineComponent({
         if (unref(resource).permissions !== undefined) {
           return unref(resource).permissions.includes(DavPermission.Updateable)
         }
-        const res = unref(resource)
-        if (isShareResource(res) && res.share?.role) {
-          return res.share.role.hasPermission(SharePermissions.update)
-        }
```

Comment:
File versions currently can't be loaded for shares (including files inside a shared folder), so this check is not needed at the moment. If needed later it should check the graph permissions instead.